engine/core/source/org/pentaho/reporting/engine/classic/core/modules/output/table/xls/ExcelReportUtil.java

```diff
@@ -243,10 +226,65 @@ public static void createXLS( final MasterReport report, final OutputStream outp
     }
 
     final FlowExcelOutputProcessor target =
-        new FlowExcelOutputProcessor( report.getConfiguration(), outputStream, report.getResourceManager() );
+      new FlowExcelOutputProcessor( report.getConfiguration(), outputStream, report.getResourceManager() );
     target.setUseXlsxFormat( false );
     final FlowReportProcessor reportProcessor = new FlowReportProcessor( report, target );
     reportProcessor.processReport();
     reportProcessor.close();
   }
+
+
+  public static void createXLS( final MasterReport report, final OutputStream outputStream,
+                                final ReportProgressListener listener )
+    throws ReportProcessingException {
+    if ( report == null ) {
+      throw new NullPointerException();
+    }
+    if ( outputStream == null ) {
+      throw new NullPointerException();
+    }
+
+    final FlowExcelOutputProcessor target =
+      new FlowExcelOutputProcessor( report.getConfiguration(), outputStream, report.getResourceManager() );
+    target.setUseXlsxFormat( false );
+    final FlowReportProcessor reportProcessor = new FlowReportProcessor( report, target );
+    if ( listener != null ) {
+      reportProcessor.addReportProgressListener( listener );
+    }
+    try {
+      reportProcessor.processReport();
+    } finally {
+      if ( listener != null ) {
+        reportProcessor.removeReportProgressListener( listener );
+      }
+      reportProcessor.close();
+    }
+  }
+
+  public static void createXLSX( final MasterReport report, final OutputStream outputStream,
+                                 final ReportProgressListener listener )
+    throws ReportProcessingException {
+    if ( report == null ) {
+      throw new NullPointerException();
+    }
+    if ( outputStream == null ) {
+      throw new NullPointerException();
+    }
+
+    final FlowExcelOutputProcessor target =
+      new FlowExcelOutputProcessor( report.getConfiguration(), outputStream, report.getResourceManager() );
+    target.setUseXlsxFormat( true );
+    final FlowReportProcessor reportProcessor = new FlowReportProcessor( report, target );
+    if ( listener != null ) {
+      reportProcessor.addReportProgressListener( listener );
+    }
+    try {
+      reportProcessor.processReport();
+    } finally {
+      if ( listener != null ) {
+        reportProcessor.removeReportProgressListener( listener );
+      }
+      reportProcessor.close();
+    }
+  }
 }
```

Comment:
Where are those two static methods used? They are most definitely not referenced in the code of this class.